virttest/libvirt_xml/base.py

```diff
@@ -320,6 +320,70 @@ def setup_attrs(self, **attrs):
                 target_obj.setup_attrs(**value)
                 setattr(self, key, target_obj)
 
+    def fetch_attrs(self):
+        """
+        Fetch attributes of xml object
+
+        :return: dict-type attributes, keys were pre-defined in xml classes
+
+        Example:
+        A dimm device 'dimm_xml' has xml like this:
+            <memory model="dimm">
+                <target>
+                    <size unit="KiB">524288</size><node>0</node>
+                </target>
+                <alias name="dimm2" />
+                <address base="0x120000000" slot="2" type="dimm" />
+            </memory>
+
+        dimm_xml.fetch_attrs()
+
+        Output should be:
+            {
+                'target': {
+                    'node': 0, 'size': 524288, 'size_unit': 'KiB'
+                    },
+                'alias': {'name': 'dimm2'},
+                'address': {
+                    'attrs': {
+                        'base': '0x120000000', 'slot': '2', 'type': 'dimm'
+                        },
+                    'type_name': 'dimm'
+                    },
+                'mem_model': 'dimm'
+            }
+
+        """
+        attrs = {}
+        slots = set(self.__all_slots__) - set(self.__uncompareable__) - {'device_tag'}
+        for key in slots:
+            try:
+                # Try to get values of each attribute, slot by slot
+                value = self[key]
```

Comment:
align with line 321 using setattr, it sounds like more proper to use get_attr() here

Reply:
@chunfuwen  Actually, no, get_attr() will easily conflict with pre-defined getter function.

Reply:
> @chunfuwen Actually, no, get_attr() will easily conflict with pre-defined getter function.

@chloerh ,here I mean use getattr(self, key).
Quto'get_attr() will easily conflict with pre-defined getter function', could you share what spcific pre-defined getter function it easily conflict with  ? since getattr is build-in function

Reply:
@chunfuwen Any xml class that has attribute named 'attr' will have a pre-defined function named 'get_attr'. 
If you still don't understand, please contact me, I'll explain more clearly face to face.

Reply:
To be more clear, 

Any xml class that has attribute named 'attr' will have **3** pre-defined functions, they are:
get_attr()
set_attr()
del_attr()

I believe it's more clear now. If not, please contact me.

Reply:
@chloerh 
I assume you means 3-pre-defined are :__getattr__(), __setattr__(), __del_attr__() rather than get_attr(), set_attr(), del_attr().
Moreover, If we use 3-pref-defined metholds, generally we call : object._getattr__('key'), but if use build-in method,it is :getattr(object, 'key'). There are clear difference.

However, It is not blocker, it is in your hand to decide whether adopt it or not